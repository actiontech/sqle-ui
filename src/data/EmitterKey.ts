enum EmitterKey {
  Refresh_User_list = 'REFRESH_USER_LIST',
  Refresh_Role_list = 'REFRESH_ROLE_LIST',
  Refresh_Whitelist_List = 'REFRESH_WHITELIST_LIST',

  Refresh_Rule_Template_List = 'REFRESH_RULE_TEMPLATE_LIST',

  Reset_Create_Order_Form = 'RESET_CREATE_ORDER_FORM',

  Reset_Test_Data_Source_Connect = 'RESET_TEST_DATA_SOURCE_CONNECT',

  Reset_Workflow_Template_Form = 'RESET_WORKFLOW_TEMPLATE_FORM',

  Rest_Audit_Plan_Form = 'REST_AUDIT_PLAN_FORM',

  Refresh_Audit_Plan_Record = 'REFRESH_AUDIT_PLAN_RECORD',
}

export default EmitterKey;
