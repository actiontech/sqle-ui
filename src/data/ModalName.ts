export enum ModalName {
  // user manage
  Add_Role = 'ADD_ROLE',
  Update_Role = 'UPDATE_ROLE',

  Add_User = 'ADD_USER',
  Update_User = 'Update_User',
  Update_User_Password = 'Update_User_Password',

  // whitelist manage
  Add_Whitelist = 'ADD_WHITELIST',
  Update_Whitelist = 'UPDATE_WHITELIST',

  // order
  Modify_Sql = 'Modify_Sql',

  // account
  Modify_Password = 'Modify_Password',

  // rule template
  Clone_Rule_Template = 'CLONE_RULE_TEMPLATE',

  // header
  SHOW_VERSION = 'SHOW_VERSION',
}
