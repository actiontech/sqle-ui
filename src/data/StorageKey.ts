enum StorageKey {
  Language = 'SQLE_LANGUAGE',
  Theme = 'SQLE_THEME',
  Token = 'SQLE_TOKEN',
  Username = 'SQLE_USERNAME',
}

export default StorageKey;
