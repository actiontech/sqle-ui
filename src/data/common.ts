export enum ResponseCode {
  SUCCESS = 0,
}

export enum SystemRole {
  admin = 'admin',
}

export enum ModalSize {
  big = 1000,
  mid = 800,
}

export const RuleListDefaultTabKey = 'ALL';
export const instanceListDefaultKey = 'all';
export const ruleTemplateListDefaultKey = 'all';

export const ModalFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 7 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 13 },
  },
};

export const PageFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 8 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 12 },
    md: { span: 10 },
  },
};

export const PageBigFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 6 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 14 },
    // md: { span: 1 },
  },
};

export const FilterFormLayout = {
  labelCol: { span: 7 },
  wrapperCol: { span: 17 },
};

export const FilterFormColLayout = {
  xs: 24,
  sm: 12,
  xl: 8,
  xxl: 6,
};

export const filterFormButtonLayoutFactory = (
  smOffset = 0,
  xlOffset = 0,
  xxlOffset = 0
) => ({
  xs: 24,
  sm: {
    span: 12,
    offset: smOffset,
  },
  xl: {
    span: 8,
    offset: xlOffset,
  },
  xxl: {
    span: 6,
    offset: xxlOffset,
  },
});

export const FilterFormRowLayout = {
  gutter: 24,
};

export enum LoginTypeEnum{
  'ldap'='ldap',
  'sqle'='sqle',
}