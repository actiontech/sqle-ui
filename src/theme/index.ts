import { makeStyles } from '@material-ui/styles';
import { Theme } from '../types/theme.type';
import darkTheme from './dark';
import lightTheme from './light';

enum SupportTheme {
  DARK = 'dark',
  LIGHT = 'light',
}

const ThemeData = {
  [SupportTheme.DARK]: darkTheme,
  [SupportTheme.LIGHT]: lightTheme,
};

const useStyles = makeStyles<Theme>((theme) => ({
  headerBg: {
    backgroundColor: theme.header.background,
    color: theme.header.color,
  },
  editor: {
    border: theme.editor.border,
    paddingRight: 1,
  },
}));

export { SupportTheme, ThemeData };
export default useStyles;
