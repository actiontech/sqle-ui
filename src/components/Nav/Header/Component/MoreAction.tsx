import { MoreOutlined } from '@ant-design/icons';
import { Dropdown, Menu } from 'antd';
import { useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { ModalName } from '../../../../data/ModalName';
import { updateNavModalStatus } from '../../../../store/nav';
import EmptyBox from '../../../EmptyBox';
import useCurrentUser from '../../../../hooks/useCurrentUser';

const MoreAction: React.FC = () => {
  const dispatch = useDispatch();
  const history = useHistory();
  const { t } = useTranslation();
  const { isAdmin } = useCurrentUser();

  const openInfoDialog = useCallback(
    (modalName: ModalName) => {
      dispatch(
        updateNavModalStatus({
          modalName: modalName,
          status: true,
        })
      );
    },
    [dispatch]
  );

  return (
    <>
      <Dropdown
        overlay={
          <Menu>
            <Menu.Item
              key="version"
              onClick={() => openInfoDialog(ModalName.SHOW_VERSION)}
            >
              {t('system.log.version')}
            </Menu.Item>

            <EmptyBox if={isAdmin}>
              <Menu.Divider />
              <Menu.Item
                key="reportStatistics"
                onClick={() => history.push('/reportStatistics')}
              >
                {t('menu.reportStatistics')}
              </Menu.Item>
              <Menu.Divider />
              <Menu.ItemGroup title={t('menu.platformManage')}>
                <Menu.Item
                  key="userCenter"
                  onClick={() => history.push('/userCenter')}
                >
                  {t('menu.userCenter')}
                </Menu.Item>
                <Menu.Item
                  key="ruleTemplate"
                  onClick={() => history.push('/rule/template')}
                >
                  {t('menu.globalRuleTemplate')}
                </Menu.Item>
                <Menu.Item
                  key="syncDataSource"
                  onClick={() => history.push('/syncDataSource')}
                >
                  {t('menu.syncDataSource')}
                </Menu.Item>
                <Menu.Item
                  key="systemSetting"
                  onClick={() => history.push('/system')}
                >
                  {t('menu.systemSetting')}
                </Menu.Item>
              </Menu.ItemGroup>
            </EmptyBox>
          </Menu>
        }
      >
        <MoreOutlined
          className="header-more-icon"
          data-testid="more-action-icon"
        />
      </Dropdown>
    </>
  );
};

export default MoreAction;
