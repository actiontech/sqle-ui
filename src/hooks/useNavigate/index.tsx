import { useCallback } from 'react';
import {
  NavigateFunction,
  NavigateOptions,
  To,
  useNavigate as useOriginNavigate,
} from 'react-router-dom';
import { SQLE_BASE_URL } from '../../data/common';

function navigateFunction(innerNavigate: NavigateFunction, delta: number): void;
function navigateFunction(
  innerNavigate: NavigateFunction,
  to: To,
  options?: NavigateOptions
): void;
function navigateFunction(
  innerNavigate: NavigateFunction,
  to: number | To,
  options?: NavigateOptions
) {
  if (typeof to === 'number') {
    return innerNavigate(to);
  }

  if (typeof to === 'string') {
    if (to.startsWith(SQLE_BASE_URL)) {
      return innerNavigate(to, options);
    }
    return innerNavigate(`${SQLE_BASE_URL}${to}`, options);
  }

  if (to.pathname?.startsWith(SQLE_BASE_URL)) {
    return innerNavigate(to, options);
  }

  return innerNavigate(
    {
      ...to,
      pathname: to.pathname ? `${SQLE_BASE_URL}${to.pathname}` : undefined,
    },
    options
  );
}

const useNavigate = (): NavigateFunction => {
  const innerNavigate = useOriginNavigate();

  const navigate = useCallback(
    (to: number | To, options?: NavigateOptions) => {
      if (typeof to === 'number') {
        navigateFunction(innerNavigate, to);
      } else {
        navigateFunction(innerNavigate, to, options);
      }
    },
    [innerNavigate]
  );

  return navigate;
};

export default useNavigate;
