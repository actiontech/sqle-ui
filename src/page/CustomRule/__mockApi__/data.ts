import { ICustomRuleResV1 } from '../../../api/common';
import { CustomRuleResV1LevelEnum } from '../../../api/common.enum';

export const customRules: ICustomRuleResV1[] = [
  {
    db_type: 'mysql',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.notice,
    rule_id: '1',
    desc: 'name1',
    rule_script: 'w+',
    type: 'rule_type1',
  },
  {
    db_type: 'oracle',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.error,
    rule_id: '2',
    desc: 'name2',
    rule_script: 'w+',
    type: 'rule_type2',
  },
];
