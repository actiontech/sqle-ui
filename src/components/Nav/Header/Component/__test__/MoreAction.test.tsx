import { fireEvent, render, screen } from '@testing-library/react';
import { ModalName } from '../../../../../data/ModalName';
import { SupportTheme } from '../../../../../theme';
import MoreAction from '../MoreAction';
import { useDispatch, useSelector } from 'react-redux';
import useNavigate from '../../../../../hooks/useNavigate';

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
  };
});

jest.mock('../../../../../hooks/useNavigate', () => jest.fn());

describe('test Nav/Header/MoreAction', () => {
  const dispatchMock = jest.fn();
  const mockNavigate = jest.fn();

  const useSelectorMock: jest.Mock = useSelector as jest.Mock;
  const useDispatchMock: jest.Mock = useDispatch as jest.Mock;
  const useNavigateMock: jest.Mock = useNavigate as jest.Mock;

  beforeEach(() => {
    useSelectorMock.mockImplementation((selector) => {
      return selector({
        user: { username: 'admin', theme: SupportTheme.LIGHT, role: 'admin' },
      });
    });
    useDispatchMock.mockImplementation(() => dispatchMock);
    useNavigateMock.mockImplementation(() => mockNavigate);
  });

  afterEach(() => {
    dispatchMock.mockClear();
    jest.clearAllMocks();
  });

  test('should match snapshot', async () => {
    const { baseElement, rerender } = render(<MoreAction />);
    fireEvent.mouseEnter(screen.getByTestId('more-action-icon'));

    await screen.findByText('system.log.version');
    expect(baseElement).toMatchSnapshot();

    useSelectorMock.mockImplementation((selector) => {
      return selector({
        user: { username: 'test', role: 'role' },
      });
    });
    rerender(<MoreAction />);

    fireEvent.mouseEnter(screen.getByTestId('more-action-icon'));

    await screen.findByText('system.log.version');
    expect(baseElement).toMatchSnapshot();
  });

  test('should opened version info modal when clicking "system.log.version"', async () => {
    render(<MoreAction />);
    fireEvent.mouseEnter(screen.getByTestId('more-action-icon'));

    await screen.findByText('system.log.version');
    expect(dispatchMock).toBeCalledTimes(0);

    fireEvent.click(screen.getByText('system.log.version'));
    expect(dispatchMock).toBeCalledTimes(1);
    expect(dispatchMock).toBeCalledWith({
      payload: { modalName: ModalName.SHOW_VERSION, status: true },
      type: 'nav/updateModalStatus',
    });
  });

  test('should jump to path when clicking menu item', async () => {
    render(<MoreAction />);

    fireEvent.mouseEnter(screen.getByTestId('more-action-icon'));

    await screen.findByText('system.log.version');

    fireEvent.click(screen.getByText('menu.reportStatistics'));
    expect(mockNavigate).nthCalledWith(1, 'reportStatistics');
    expect(mockNavigate).toBeCalledTimes(1);

    fireEvent.click(screen.getByText('menu.userCenter'));
    expect(mockNavigate).nthCalledWith(2, 'userCenter');
    expect(mockNavigate).toBeCalledTimes(2);

    fireEvent.click(screen.getByText('menu.ruleManager'));
    expect(mockNavigate).nthCalledWith(3, 'rule/template');
    expect(mockNavigate).toBeCalledTimes(3);

    fireEvent.click(screen.getByText('menu.systemSetting'));
    expect(mockNavigate).nthCalledWith(4, 'system');
    expect(mockNavigate).toBeCalledTimes(4);
  });
});
