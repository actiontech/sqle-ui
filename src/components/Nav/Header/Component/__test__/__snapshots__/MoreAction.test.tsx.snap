// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Nav/Header/MoreAction should match snapshot 1`] = `
<body>
  <div>
    <span
      aria-label="more"
      class="anticon anticon-more ant-dropdown-trigger header-more-icon ant-dropdown-open"
      data-testid="more-action-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </div>
  <div
    style="position: absolute; top: 0px; left: 0px; width: 100%;"
  >
    <div>
      <div
        class="ant-dropdown"
        style="opacity: 0; pointer-events: none;"
      >
        <ul
          class="ant-dropdown-menu ant-dropdown-menu-root ant-dropdown-menu-vertical ant-dropdown-menu-light"
          data-menu-list="true"
          role="menu"
          tabindex="0"
        >
          <li
            class="ant-dropdown-menu-item ant-dropdown-menu-item-only-child"
            data-menu-id="rc-menu-uuid-test-version"
            role="menuitem"
            tabindex="-1"
          >
            <span
              class="ant-dropdown-menu-title-content"
            >
              system.log.version
            </span>
          </li>
          <li
            class="ant-dropdown-menu-item-divider"
          />
          <li
            class="ant-dropdown-menu-item ant-dropdown-menu-item-only-child"
            data-menu-id="rc-menu-uuid-test-reportStatistics"
            role="menuitem"
            tabindex="-1"
          >
            <span
              class="ant-dropdown-menu-title-content"
            >
              menu.reportStatistics
            </span>
          </li>
          <li
            class="ant-dropdown-menu-item-divider"
          />
          <li
            class="ant-dropdown-menu-item-group"
          >
            <div
              class="ant-dropdown-menu-item-group-title"
              title="menu.platformManage"
            >
              menu.platformManage
            </div>
            <ul
              class="ant-dropdown-menu-item-group-list"
            >
              <li
                class="ant-dropdown-menu-item-group"
              >
                <div
                  class="ant-dropdown-menu-item-group-title"
                  title="menu.userCenter"
                >
                  menu.userCenter
                </div>
                <ul
                  class="ant-dropdown-menu-item-group-list"
                >
                  <li
                    class="ant-dropdown-menu-item ant-dropdown-menu-item-only-child"
                    data-menu-id="rc-menu-uuid-test-user"
                    role="menuitem"
                    tabindex="-1"
                  >
                    <span
                      class="ant-dropdown-menu-title-content"
                    >
                      menu.user
                    </span>
                  </li>
                  <li
                    class="ant-dropdown-menu-item ant-dropdown-menu-item-only-child"
                    data-menu-id="rc-menu-uuid-test-role"
                    role="menuitem"
                    tabindex="-1"
                  >
                    <span
                      class="ant-dropdown-menu-title-content"
                    >
                      menu.role
                    </span>
                  </li>
                  <li
                    class="ant-dropdown-menu-item ant-dropdown-menu-item-only-child"
                    data-menu-id="rc-menu-uuid-test-userGroup"
                    role="menuitem"
                    tabindex="-1"
                  >
                    <span
                      class="ant-dropdown-menu-title-content"
                    >
                      menu.userGroup
                    </span>
                  </li>
                </ul>
              </li>
              <li
                class="ant-dropdown-menu-item ant-dropdown-menu-item-only-child"
                data-menu-id="rc-menu-uuid-test-ruleTemplate"
                role="menuitem"
                tabindex="-1"
              >
                <span
                  class="ant-dropdown-menu-title-content"
                >
                  menu.ruleTemplate
                </span>
              </li>
              <li
                class="ant-dropdown-menu-item ant-dropdown-menu-item-only-child"
                data-menu-id="rc-menu-uuid-test-systemSetting"
                role="menuitem"
                tabindex="-1"
              >
                <span
                  class="ant-dropdown-menu-title-content"
                >
                  menu.systemSetting
                </span>
              </li>
            </ul>
          </li>
        </ul>
        <div
          aria-hidden="true"
          style="display: none;"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`test Nav/Header/MoreAction should match snapshot 2`] = `
<body>
  <div>
    <span
      aria-label="more"
      class="anticon anticon-more ant-dropdown-trigger header-more-icon ant-dropdown-open"
      data-testid="more-action-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </div>
  <div
    style="position: absolute; top: 0px; left: 0px; width: 100%;"
  >
    <div>
      <div
        class="ant-dropdown"
        style="opacity: 0; pointer-events: none;"
      >
        <ul
          class="ant-dropdown-menu ant-dropdown-menu-root ant-dropdown-menu-vertical ant-dropdown-menu-light"
          data-menu-list="true"
          role="menu"
          tabindex="0"
        >
          <li
            class="ant-dropdown-menu-item ant-dropdown-menu-item-only-child"
            data-menu-id="rc-menu-uuid-test-version"
            role="menuitem"
            tabindex="-1"
          >
            <span
              class="ant-dropdown-menu-title-content"
            >
              system.log.version
            </span>
          </li>
          <li
            class="ant-dropdown-menu-item-divider"
          />
          <li
            class="ant-dropdown-menu-item ant-dropdown-menu-item-only-child"
            data-menu-id="rc-menu-uuid-test-reportStatistics"
            role="menuitem"
            tabindex="-1"
          >
            <span
              class="ant-dropdown-menu-title-content"
            >
              menu.reportStatistics
            </span>
          </li>
          <li
            class="ant-dropdown-menu-item-divider"
          />
        </ul>
        <div
          aria-hidden="true"
          style="display: none;"
        />
      </div>
    </div>
  </div>
</body>
`;
